os/arch/arm/src/amebad/amebad_rtc_lowerhalf.c

```diff
@@ -0,0 +1,463 @@
+	ts.tv_nsec = 0;
+
+	/* Now set the time (to one second accuracy) */
+
+	return up_rtc_settime(&ts);
+}
+
+/****************************************************************************
+ * Name: amebad_setalarm
+ *
+ * Description:
+ *   Set a new alarm.  This function implements the setalarm() method of the
+ *   RTC driver interface
+ *
+ * Input Parameters:
+ *   lower     - A reference to RTC lower half driver state structure
+ *   alarminfo - Provided information needed to set the alarm
+ *
+ * Returned Value:
+ *   Zero (OK) is returned on success; a negated errno value is returned
+ *   on any failure.
+ *
+ ****************************************************************************/
+
+#ifdef CONFIG_RTC_ALARM
+static int amebad_setalarm(FAR struct rtc_lowerhalf_s *lower, FAR const struct lower_setalarm_s *alarminfo)
+{
+	FAR struct amebad_lowerhalf_s *rtc;
+	int ret;
+
+	DEBUGASSERT(lower != NULL && alarminfo != NULL && alarminfo->id == 0);
+	rtc = (FAR struct amebad_lowerhalf_s *)lower;
+
+	/* Get exclusive access to the alarm */
+
+	ret = sem_wait(&rtc->devsem);
+	if (ret < 0) {
+		return ret;
+	}
+
+	g_amebad_alarm.hour = alarminfo->time.tm_hour;
+	g_amebad_alarm.yday = alarminfo->time.tm_yday;
+	g_amebad_alarm.min = alarminfo->time.tm_min;
+	g_amebad_alarm.sec = alarminfo->time.tm_sec;
+
+	ret = -EINVAL;
+	if (alarminfo->id == 0) {
+		struct timespec ts;
+
+		/* Convert the RTC time to a timespec (1 second accuracy) */
+
+		ts.tv_sec = mktime((FAR struct tm *)&alarminfo->time);
+		ts.tv_nsec = 0;
+
+		/* Remember the callback information */
+
+		rtc->cb = alarminfo->cb;
+		rtc->priv = alarminfo->priv;
+
+		/* And set the alarm */
+		ret = rtc_set_alarm(&g_amebad_alarm, (alarm_irq_handler)amebad_alarm_callback);
+		if (ret < 0) {
+			rtc->cb = NULL;
+			rtc->priv = NULL;
+		}
+	}
+
+	sem_post(&rtc->devsem);
+	return ret;
+}
+#endif
+
+/****************************************************************************
+ * Name: amebad_setrelative
+ *
+ * Description:
+ *   Set a new alarm relative to the current time.  This function implements
+ *   the setrelative() method of the RTC driver interface
+ *
+ * Input Parameters:
+ *   lower - A reference to RTC lower half driver state structure
+ *   alarminfo - Provided information needed to set the alarm
+ *
+ * Returned Value:
+ *   Zero (OK) is returned on success; a negated errno value is returned
+ *   on any failure.
+ *
+ ****************************************************************************/
+
+#ifdef CONFIG_RTC_ALARM
+static int amebad_setrelative(FAR struct rtc_lowerhalf_s *lower, FAR const struct lower_setrelative_s *alarminfo)
+{
+	FAR struct amebad_lowerhalf_s *rtc;
+	FAR struct timespec ts;
+	int ret = -EINVAL;
+
+	DEBUGASSERT(lower != NULL && alarminfo != NULL && alarminfo->id == 0);
+	rtc = (FAR struct amebad_lowerhalf_s *)lower;
+
+	/* Get exclusive access to the alarm */
+
+	ret = sem_wait(&rtc->devsem);
+	if (ret < 0) {
+		printf("ERROR: sem_wait failed: %d\n", ret);
```

Comment:
I think we'd better use proper dbg macro instead of printf.

Reply:
I deleted it to keep it being the same as setalarm.